### DISCUSSION TITLE
Add some predefined templates

### DISCUSSION THREAD

**Discussion by sovetski** [saadeghi/daisyui#2940] [Ideas / feature requests]
Hi,

First, a big thanks for this great project!

For me the most important thing is missing, which is an example of "templates" which are created by combining some components.

Yes, there is a store where you sell some templates, but it is for a complete website templates, as a developer we always need same structure on 80% of websites. I mean, a navbar with footer and some content, a sidebar optionally etc.

On Mantine for example, we have this: https://mantine.dev/app-shell/?e=BasicAppShell

Where the developer added some possibilities.

Is it possible to do the same thing, guys?

To just don't ask and do something, I created this example: https://play.tailwindcss.com/8baSdLqPFP

Is it OK to add this (or improved version or other ones) on the official documentation? Maybe a new section named "Showcase"?

Big screen:
![image](https://github.com/saadeghi/daisyui/assets/13520683/9c4082d9-643e-4e60-8cca-ac1702143bca)

Medium:
![image](https://github.com/saadeghi/daisyui/assets/13520683/a475c431-0aaa-403c-926a-703c7494a210)

Mobile:
![image](https://github.com/saadeghi/daisyui/assets/13520683/0766bce0-5f7e-4541-b75f-e4471ce307cf)

Thanks in advance

**Comment by saadeghi** [comment]
Honestly I don't want to provide a big chunk of HTML code for different layouts. It's something you can find anywhere else and even if I put these examples on daisyUI website, it's not daisyUI. It's mostly grid and spacing utility class names (with some daisyUI components in them)

I've been thinking about an idea of providing short class names for common grid layouts that a website may need. For example using a class name like `layout-4col` and it gives a grid of 4 columns but automatically shows a grid of `3`, `2`, `1` on smaller screens.  

The idea has many challenges to get into production. For example the class names could be so abstract and weird if it has to be descriptive.  

Like `layout-1col-3col-2col` 🤢
```
┌─────────────────┐
│                 │
├─────┬─────┬─────┤
│     │     │     │
├─────┴──┬──┴─────┤
│        │        │
└────────┴────────┘
```
Or `layout-1col-2narrowcoland1widecol` 🤮
```
┌─────────────────┐
│                 │
├──┬──┬───────────┤
│  │  │           │
│  │  │           │
└──┴──┴───────────┘
```

I also thought about simple put numbers for each layout and not make the class name descriptive (like `layout-1`, `layout-2`,…) but I'm still not sure.

**Reply by sovetski** [reply]
I agree with you, having long class names that aren't very intuitive can be complicated.

Personally, I like the idea of simply numbering them, for example just "layout-1", "layout-2", "layout-3", etc. The only issue with that is when looking at the class names, you can't tell what they are, you'd have to look at the documentation or inspect the elements to see what they look like.

What do you think of this? "layout-[h,m,f]"? Instead of numbers, adding letters which mean "h - header", "m - main", "f - footer"? And perhaps also "s - sidebar" and we can use it like "layout-[h,m,s,f]". It's still a matter of taste and preference in the end, each with its own advantages and disadvantages. But in any case, I don't think there's a need to provide all possible layouts, just those that are almost always used, like having a header, footer, etc.

**Comment by adenta** [comment]
curious if in the past year, any standards evolved here
